src/FakeItEasy/Creation/CreationResult.cs

```diff
@@ -21,73 +23,83 @@ private CreationResult(object result)
+
+        public static CreationResult FailedToCreateDummy(Type type, List<ResolvedConstructor> consideredConstructors) =>
+            new CreationResult(type, CreationMode.Dummy, consideredConstructors: consideredConstructors);
+
+        public static CreationResult FailedToCreateFake(Type type, string reasonForFailure) =>
+            new CreationResult(type, CreationMode.Fake, reasonsForFailure: new List<string> { reasonForFailure });
+
+        public static CreationResult FailedToCreateFake(Type type, List<ResolvedConstructor> consideredConstructors) =>
+            new CreationResult(type, CreationMode.Fake, consideredConstructors: consideredConstructors);
+
+        /// <summary>
+        /// Returns a creation result for a dummy by combining two results.
+        /// Successful results are preferred to failed. Failed results will have their reasons for failure aggregated.
+        /// This method may alter either or both of its arguments.
+        /// </summary>
+        /// <param name="one">One result to merge. May be <code>null</code>.</param>
+        /// <param name="other">The other result to merge. Must not be <code>null</code>.</param>
+        /// <returns>A combined creation result. Successful if either input was successful, and failed otherwise.</returns>
+        public static CreationResult MergeIntoDummyResult(CreationResult one, CreationResult other)
         {
-            if (one == null)
-            {
-                return other;
-            }
+            Guard.AgainstNull(other, "other");
 
-            if (other == null || one.WasSuccessful)
+            if (one == null || other.WasSuccessful)
             {
-                return one;
+                return AsDummyResult(other);
             }
 
-            if (other.WasSuccessful)
+            if (one.WasSuccessful)
             {
-                return other;
+                return AsDummyResult(one);
             }
 
             return new CreationResult(
                 one.type,
+                CreationMode.Dummy,
                 MergeReasonsForFailure(one.reasonsForFailure, other.reasonsForFailure),
                 MergeConsideredConstructors(one.consideredConstructors, other.consideredConstructors));
         }
 
-        public object GetResultAsFake()
+        private static CreationResult AsDummyResult(CreationResult original)
         {
-            if (this.WasSuccessful)
-            {
-                return this.result;
-            }
-
-            throw new FakeCreationException(this.GetFailedToCreateResultMessage("fake"));
-        }
-
-        public object GetResultAsDummy()
-        {
-            if (this.WasSuccessful)
-            {
-                return this.result;
-            }
-
-            throw new DummyCreationException(this.GetFailedToCreateResultMessage("dummy"));
+            original.creationMode = CreationMode.Dummy;
+            return original;

```

Comment:
I would expect such a method to create a new object rather than modify the original. Maybe it's overkill, but immutability does help reducing bugs.

Reply:
Will do.